pkg/controller/database/rdsinstance.go

```diff
@@ -0,0 +1,243 @@
+// controller and its corresponding reconciler to the manager with any runtime configuration.
+type RDSInstanceController struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *RDSInstanceController) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1beta1.RDSInstanceGroupVersionKind),
+		resource.WithExternalConnecter(&connector{
+			kube:        mgr.GetClient(),
+			newClientFn: rds.NewClient,
+		}))
+
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1beta1.RDSInstanceKind, v1beta1.Group))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1beta1.RDSInstance{}).
+		Complete(r)
+}
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(credentials []byte, region string) (rds.Client, error)
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mg.(*v1beta1.RDSInstance)
+	if !ok {
+		return nil, errors.New(errNotRDSInstance)
+	}
+
+	p := &awsv1alpha2.Provider{}
+	if err := c.kube.Get(ctx, meta.NamespacedNameOf(cr.Spec.ProviderReference), p); err != nil {
+		return nil, errors.Wrap(err, errGetProvider)
+	}
+
+	s := &corev1.Secret{}
+	n := types.NamespacedName{Namespace: p.Spec.Secret.Namespace, Name: p.Spec.Secret.Name}
+	if err := c.kube.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+
+	rdsClient, err := c.newClientFn(s.Data[p.Spec.Secret.Key], p.Spec.Region)
+	return &external{client: rdsClient, kube: c.kube}, errors.Wrap(err, errCreateRDSClient)
+}
+
+type external struct {
+	client rds.Client
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mg.(*v1beta1.RDSInstance)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errNotRDSInstance)
+	}
+	// TODO(muvaf): There are some parameters that require a specific call
+	// for retrieval. For example, DescribeDBInstancesOutput does not expose
+	// the tags map of the RDS instance, you have to make ListTagsForResourceRequest
+	req := e.client.DescribeDBInstancesRequest(&awsrds.DescribeDBInstancesInput{DBInstanceIdentifier: aws.String(meta.GetExternalName(cr))})
+	req.SetContext(ctx)
+	rsp, err := req.Send()
+	if err != nil {
+		return resource.ExternalObservation{}, errors.Wrap(resource.Ignore(rds.IsErrorNotFound, err), errDescribeFailed)
+	}
+
+	// Describe requests can be used with filters, which then returns a list.
+	// But we use an explicit identifier, so, if there is no error, there should
+	// be only 1 element in the list.
+	instance := rsp.DBInstances[0]
+	current := cr.Spec.ForProvider.DeepCopy()
+	rds.LateInitialize(&cr.Spec.ForProvider, &instance)
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
```

Comment:
No changes needed, but I suspect we can drop this check in future. I'm pretty sure `Update` is a no-op if nothing about the resource has changed.